src/decoder.rs

```diff
@@ -294,36 +294,47 @@ fn build_type_id_search_option(type_id_args: [u8; 32]) -> CellQueryOptions {
 
 fn build_batch_search_options(
     type_args: [u8; 32],
-    avaliable_code_hashes: &[H256],
+    available_script_ids: &[ScriptId],
 ) -> Vec<CellQueryOptions> {
-    avaliable_code_hashes
+    available_script_ids
         .iter()
-        .map(|code_hash| {
-            let type_script = Script::new_builder()
-                .code_hash(code_hash.0.pack())
-                .hash_type(ScriptHashType::Data1.into())
-                .args(type_args.to_vec().pack())
-                .build();
-            CellQueryOptions::new_type(type_script)
-        })
+        .map(
+            |ScriptId {
+                 code_hash,
+                 hash_type,
+             }| {
+                let hash_type: ScriptHashType = hash_type.into();
+                let type_script = Script::new_builder()
+                    .code_hash(code_hash.0.pack())
+                    .hash_type(hash_type.into())
+                    .args(type_args.to_vec().pack())
+                    .build();
+                CellQueryOptions::new_type(type_script)
+            },
+        )
         .collect()
 }
 
-pub(crate) fn decode_spore_data(spore_data: &[u8]) -> Result<SporeContentField, Error> {
+pub(crate) fn decode_spore_data(spore_data: &[u8]) -> Result<(Value, String), Error> {
+    if spore_data[0] == 0u8 {
+        let dna = hex::encode(&spore_data[1..]);
```

Comment:
what this first `byte` means to?

Reply:
It helps data parser to distinguish raw bytes from UTF-8 JSON bytes. We parse the DNA differently when the first byte is 0, '"', '[' or '{' respectively.